docs/data/data-grid/prompt/PromptWithExamples.tsx

```diff
@@ -0,0 +1,94 @@
+import * as React from 'react';
+import {
+  DataGridPremium,
+  Unstable_GridToolbarPromptControl as GridToolbarPromptControl,
+  GridToolbar,
+  GridToolbarContainer,
+} from '@mui/x-data-grid-premium';
+import {
+  mockPromptResolver,
+  randomBoolean,
+  randomCompanyName,
+  randomCountry,
+  randomCreatedDate,
+  randomEmail,
+  randomInt,
+  randomJobTitle,
+  randomPhoneNumber,
+  randomTraderName,
+  useDemoData,
+} from '@mui/x-data-grid-generator';
+import Stack from '@mui/material/Stack';
+
+function ToolbarWithPromptInput() {
+  return (
+    <GridToolbarContainer sx={{ p: 1 }}>
+      <Stack flex={1} gap={1}>
+        <GridToolbar />
+        <GridToolbarPromptControl onPrompt={mockPromptResolver} />
+      </Stack>
+    </GridToolbarContainer>
```

Comment:
Minor suggestion to improve the spacing:

```suggestion
    <Stack flex={1} gap={0.5} sx={{ px: 0.5 }}>
      <GridToolbar />
      <Box sx={{ px: 0.5 }}>
        <GridToolbarPromptControl onPrompt={mockPromptResolver} />
      </Box>
    </Stack>
```

(we don't need the `GridToolbarContainer`, as `GridToolbar` already renders it.